packages/loader/container-loader/src/container.ts

```diff
@@ -291,6 +294,17 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private _parentBranch: string | null = null;
     private _connectionState = ConnectionState.Disconnected;
     private readonly _audience: Audience;
+    /**
+     * The code details that were used to load the FluidModule in our current ContainerContext, or undefined if we
+     * don't have a context loaded.
+     */
+    private _codeDetails: IFluidCodeDetails | undefined;
+
+    /**
+     * The Fluid module that we loaded in our current ContainerContext, or undefined if we don't have a context
+     * loaded.
+     */
+    private _fluidModule: IFluidModule | undefined;
```

Comment:
i really preferred these on the context to reduce mutable state. While i get the desire to reduce concepts, the container context is by definition needs some definition of runtime code, so i'm not totally clear what we gain here.

Reply:
This doesn't change the quantity of mutable state, it just relocates the ownership up the stack.  So e.g. this `_fluidModule` member replaces the `fluidModuleP` from the ContainerContext

Reply:
it's super important the codeDetail matches the codeDetails of the context, which can diverge from the quorum in the case where were skipped reload due to a new and compatible codeDetails, this change makes that easier to diverge. I also don't necessarily see the issue of it existing on the context. Overall i think we are trying to minimize the amount work done by the runtime, which is the only thing that should ever have access to the context.

Reply:
Related: https://github.com/microsoft/FluidFramework/pull/4122#discussion_r513220479